app/models/manageiq/providers/container_manager.rb

```diff
@@ -19,6 +19,7 @@ class ContainerManager < BaseManager
     has_many :container_build_pods, :foreign_key => :ems_id, :dependent => :destroy
     has_many :container_templates, :foreign_key => :ems_id, :dependent => :destroy
     has_one :container_deployment, :foreign_key => :deployed_ems_id, :inverse_of => :deployed_ems
+    has_many :miq_alert_statuses, :through => :container_nodes
```

Comment:
Yea, I've got nothing here.
Either associate all alert statuses directly to the container (or container_node) or modify the table to add a `manager_id` - even if you could define this in an active record way, the performance would be terrible.